### ISSUE TITLE
Inconsistent use of total vs kinetic energy

### ISSUE THREAD

**Issue by tdealtry** [WCSim/WCSim#184]
In WCSimPrimaryGeneratorAction.cc, there is inconsistency between what input particle "energy" means.
* muline == kinetic energy
* (old muline == total energy)
* normal == total energy
* laser == total energy

To be unambiguous, if you set the particle gun with `/gun/energy 5 MeV`, you get a 5.511 total (5 kinetic) MeV particle in the output file. If instead you use a muline line `$ track 11 5 0.48280 0.79725 -0.36235 0`, you get a 5 total (4.489 kinetic) MeV particle in the output file.
(I've taken total energy to mean `WCSimRootTrack::GetE()`, and kinetic energy to mean `WCSimRootTrack::GetE() - WCSimRootTrack::GetM()`)

This is confusing. A convention should be decided on, made clear, and made consistent.

**Comment by tdealtry** [comment]
I'd favour making kinetic energy the variable used as input.
Running muline with a 0 MeV input, Geant4 creates a primary particle in `G4ParticleGun::GeneratePrimaryVertex()` with `Kinetic energy -0.511 Total energy 0`. But WCSim saves it as `Kinetic energy 0 Total energy 0.511`. Which is the same as if you input 0.511 MeV energy. Something very weird is going on... checks for negative inputs need to be done at the very least.

**Comment by cwwalter** [comment]
The muline and old muline are very old (from pre HK days and should not be used).  In fact I thought we already removed them :)

I would like to keep it as total energy for a few reasons.  One of them is that the input faithfully follows the nuance format (also used by some other collaborations) and the input in that specification is defined in total energy.

**Comment by tdealtry** [comment]
Thanks Chris, that's understandable. In that case, I guess there are a few things to do to avoid confusion

1. In the short term, make it clear what the definition of energy is in each case. Add some handling of negative energy too. Also mention that the vecfile option shouldn't be used. These should be added to the .mac file, documentation PDF, and MakeKin.py
2. Add examples of how to use the gps commands to do the same things as MakeKin.py does. For example:
  * isotropic particle directions
  * particles pointed towall
  * vertices distributed uniformly around the tank

3. Remove the ability to use the muline formats. But before this happens, clarify how neutrino event generators will provide events to WCSim

**Comment by tdealtry** [comment]
Do these sound sensible?

**Comment by cwwalter** [comment]
> In the short term, make it clear what the definition of energy is in each case. Add some handling of negative energy too. Also mention that the vecfile option shouldn't be used. These should be added to the .mac file, documentation PDF, and MakeKin.py

Sure. More documentation is good.  Of course the main input is through the vector file.  So what we really want is a good documentation of that format.  There is a link to it in the wiki now:

http://neutrino.phy.duke.edu/nuance-format/

but we could move this into the repo itself somewhere.

> Add examples of how to use the gps commands to do the same things as MakeKin.py does. For example:
>

OK. I'll let Erin oversee this when she is available in a few months.

> Remove the ability to use the muline formats. But before this happens, clarify how neutrino event generators will provide events to WCSim

Right.  So, this is clear.  NUANCE format files.

**Comment by tdealtry** [comment]
I've just realised I was confused. The "muline" option I listed above is actually "Nuance". `WCSimPrimaryGeneratorAction` needs cleaning up - nuance is listed as a type of muline format (and the old muline is hardcoded off). Perhaps the old muline option should be dropped entirely, and the "muline" terminology replaced with "nuance"?

But now I understand what nuance is, MakeKin.py actually creates these files. So gps equivalents are less important.

**Comment by cwwalter** [comment]
Yes, muline should be completely removed.  It's not actually really nuance.  It takes single line muons in nuance line format but doesn't require an entire event.  It is one of the very first inputs I wrote when the code was originally being put together for the T2K 2KM detector.  It is copying something skdetsim at least used to be able to do.
